tracing-core/src/subscriber.rs

```diff
@@ -564,6 +564,37 @@ impl Interest {
     }
 }
 
+/// A no-op [`Subscriber`]
+///
+/// [`NoSubscriber`] implements the [`Subscriber`] trait by never being enabled,
+/// never being interested in any callsite, and drops all spans and events.
+#[derive(Debug, Copy, Clone)]
+pub struct NoSubscriber;
```

Comment:
I don't _think_ this will ever need to have fields. But, for forwards-compatibility purposes, the most cautious thing to do might be to add a single private field and a `Default` implementation, just in case we ever need to add a field.

I can't really imagine a scenario where we would have to add fields to this, but it may be worth hedging our bets nonetheless.

Reply:
Done.

Reply:
I guess there's the question of whether it should be `Copy`, if it might stop being `Copy` in future.